### ISSUE TITLE
Bar transition oddities

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.js#4251]
In this pen which is in "layout first" transition ordering, there seems to be an extra bar-height jump: https://codepen.io/nicolaskruchten/pen/WNNeZgz?editors=0010

**Comment by etpinard** [comment]
![Peek 2019-10-07 14-27](https://user-images.githubusercontent.com/6675409/66338218-ac3cf200-e90e-11e9-8301-5bfa3d840224.gif)



Hmm. Interesting, scatter seems to behave the same: https://codepen.io/etpinard/pen/dyyPvvZ?editors=0010

![Peek 2019-10-07 14-28](https://user-images.githubusercontent.com/6675409/66338248-bd85fe80-e90e-11e9-9039-8d872040e410.gif)

**Comment by etpinard** [comment]
... and by the way, `transition.ordering: 'layout first'` is the default:

https://github.com/plotly/plotly.js/blob/bead43ff7d5cbcedfd2bf29c25cf01b1b4074df9/src/plots/animation_attributes.js#L123-L133

**Comment by nicolaskruchten** [comment]
Yep, I know, I was just calling attention to it and making it easier to change to the non-default options in the pen ;)

**Comment by etpinard** [comment]
After some investigations, it doesn't look like `bar/plot.js` (cc @antoinerg ) is doing anything that results in any discrepancies between bar and scatter transitions.

Layout + traces transitions don't look very good at the moment, but they don't look any worse than what was proposed in https://github.com/plotly/plotly.js/pull/3217 

So, this ticket is essentially a duplicate of https://github.com/plotly/plotly.js/issues/1687

**Comment by nicolaskruchten** [comment]
I agree in principle, although solving the double-jump issue seems easier than solving #1687 altogether... what does that first jump represent? The middle point/bar jumps from 4 to 5, then the yaxis range smoothly transitions along with the point/bar, then the point/bar jumps to 6. I would expect that under layout-first the first jump wouldn't happen, and the point/bar would jump from 4 to 6... ?

**Comment by etpinard** [comment]
> although solving the double-jump issue seems easier than solving

Double jump? In the gif above I can only detect one "jump"

> what does that first jump represent?

Here's the current logic under `transition.ordering: 'layout first'`:

https://github.com/plotly/plotly.js/blob/bead43ff7d5cbcedfd2bf29c25cf01b1b4074df9/src/plots/plots.js#L2617-L2621

- We first transition the y-axis range from ~ `[3,5]` to ~ `[3,6]` while not updating the trace position
- Then, we update the trace position instantaneously from `y=[3,4,5]` to `y=[3,6,5]` (aka the "jump")

**Comment by nicolaskruchten** [comment]
You don’t see the immediate jump 4-5, then slide, then jump 5-6?

**Comment by etpinard** [comment]
... so w/o the "jump" at the end of the transition we'd get: the correct y-axis range of ~ `[3,6]` but the trace would have coordinates `y=[3,4,5]`.

**Comment by nicolaskruchten** [comment]
Yeah the jump at the end is fine. This issue is about the first jump :)

**Comment by etpinard** [comment]
Ah now I see it, there's a race condition in:

https://github.com/plotly/plotly.js/blob/bead43ff7d5cbcedfd2bf29c25cf01b1b4074df9/src/plots/plots.js#L2617-L2621

Sorry for the confusion!

Codepen attempt: https://codepen.io/etpinard/pen/oNNgGem?editors=0010

**Comment by etpinard** [comment]
Scatter version: https://codepen.io/etpinard/pen/NWWPaaR?editors=0010

**Comment by nicolaskruchten** [comment]
phew! Those pens look like what I would expect, yep!
